### PR TITLE
Privacy scanner types

### PR DESCRIPTION
**Pull request by pownkel**
#### Details

Define the API interface and storage document types for privacy scanner

##### Motivation

Create data types that will be used for the privacy scanner

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)
- [ ] Validated in an Azure resource group

### CODE REVIEW

File: packages/service-library/src/web-api/api-contracts/privacy-scan-types/privacy-result-response.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { DeepScanResultItem, ScanReport, ScanRun } from '../scan-result-response';
+
+export type privacyResultResponse = {
```
Comment (lamaks) [inline packages/service-library/src/web-api/api-contracts/privacy-scan-types/privacy-result-response.ts:6]: no need for this type, we will use existing structure with `privacyScan?: boolean;` flag

File: packages/service-library/src/web-api/api-contracts/privacy-scan-types/privacy-scan-run-request.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { PrivacyScanConfiguration, StorageDocument } from 'storage-documents';
+import { ReportGroup, Website } from '../scan-run-request';
+
+export type PrivacyScanConfigDefinition = Omit<PrivacyScanConfiguration, keyof StorageDocument>;
+
+export interface PrivacyScanRequest {
```
Comment (lamaks) [inline packages/service-library/src/web-api/api-contracts/privacy-scan-types/privacy-scan-run-request.ts:9]: no need for this type, we will use existing structure with `privacyScan?: boolean;` flag

File: packages/storage-documents/src/item-type.ts [hunk 3]
```diff
@@ -8,4 +8,8 @@ export enum ItemType {
     onDemandPageScanRunResult = 'pageScanRunResult',
     websiteScanResult = 'websiteScanResult',
     websiteScanResultPart = 'websiteScanResultPart',
+    privacyPageScanBatchRequest = 'privacyPageScanBatchRequest',
+    privacyPageScanRequest = 'privacyPageScanRequest',
+    privacyScanRunResult = 'privacyScanRunResult',
+    privacyScanConfiguration = 'privacyScanConfiguration',
```
Comment (lamaks) [inline packages/storage-documents/src/item-type.ts:14]: no need for this type, we will use existing structure

File: packages/storage-documents/src/privacy-scan-types/privacy-page-scan-batch-request.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    privacyScanConfig: PrivacyScanConfiguration;
+}
+
+export interface PrivacyScanRunBatchRequest {
+    scanId: string;
+    url: string;
+    site?: WebsiteRequest;
+    priority: number;
+    reportGroups?: ReportGroupRequest[];
+}
```
Comment (lamaks) [inline packages/storage-documents/src/privacy-scan-types/privacy-page-scan-batch-request.ts:24]: no need for this type, we will use existing structure with `privacyScan?: boolean;` flag

File: packages/storage-documents/src/privacy-scan-types/privacy-page-scan-request.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ItemType } from '../item-type';
+import { StorageDocument } from '../storage-document';
+import { WebsiteRequest, ReportGroupRequest } from '../on-demand-page-scan-batch-request';
+
+export interface PrivacyPageScanRequest extends StorageDocument {
```
Comment (lamaks) [inline packages/storage-documents/src/privacy-scan-types/privacy-page-scan-request.ts:8]: no need for this type, we will use existing structure with `privacyScan?: boolean;` flag

File: packages/storage-documents/src/privacy-scan-types/privacy-page-scan-result.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { OnDemandPageScanReport, OnDemandPageScanRunResult, WebsiteScanRef } from '../on-demand-page-scan-result';
+import { ItemType } from '../item-type';
+import { StorageDocument } from '../storage-document';
+
+export interface PrivacyPageScanResult extends StorageDocument {
```
Comment (lamaks) [inline packages/storage-documents/src/privacy-scan-types/privacy-page-scan-result.ts:8]: no need for this type, we will use existing structure with `privacyScan?: boolean;` flag

File: packages/storage-documents/src/privacy-scan-types/privacy-scan-configuration.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ItemType, StorageDocument } from '..';
+
+// Data type stored by WCP
+export interface PrivacyScanConfiguration extends StorageDocument {
```
Comment (lamaks) [inline packages/storage-documents/src/privacy-scan-types/privacy-scan-configuration.ts:7]: do we need to have unique configuration per each scan? Will be sufficient to have privacy scan configuration defined in a code as a json file or similar?

Comment (pownkel) [reply packages/storage-documents/src/privacy-scan-types/privacy-scan-configuration.ts]: I discussed with Udaya, we can use a common config for the MVP but may eventually need to add options for sites with custom banners. I'll remove this file for now.

**Pull request merged**